Tests/SwiftProtobufTests/Test_JSONEncodingOptions.swift

```diff
@@ -0,0 +1,104 @@
+// Tests/SwiftProtobufTests/Test_JSONEncodingOptions.swift - Various JSON tests
+//
+// Copyright (c) 2014 - 2018 Apple Inc. and the project authors
+// Licensed under Apache License v2.0 with Runtime Library Exception
+//
+// See LICENSE.txt for license information:
+// https://github.com/apple/swift-protobuf/blob/master/LICENSE.txt
+//
+// -----------------------------------------------------------------------------
+///
+/// Test for the use of JSONEncodingOptions
+///
+// -----------------------------------------------------------------------------
+
+import Foundation
+import XCTest
+import SwiftProtobuf
+
+class Test_JSONEncodingOptions: XCTestCase {
+
+  func testAlwaysPrintEnumsAsInts() {
+    // Use explict options (the default is false), just to be pedantic.
+    var asStrings = JSONEncodingOptions()
+    asStrings.alwaysPrintEnumsAsInts = false
+    var asInts = JSONEncodingOptions()
+    asInts.alwaysPrintEnumsAsInts = true
+
+    // Toplevel fields
+
+    let msg1 = ProtobufUnittest_Message3.with {
+      $0.optionalEnum = .bar
+    }
+    XCTAssertEqual(try msg1.jsonString(options: asStrings), "{\"optionalEnum\":\"BAR\"}")
+    XCTAssertEqual(try msg1.jsonString(options: asInts), "{\"optionalEnum\":1}")
+
+    let msg2 = ProtobufUnittest_Message3.with {
+      $0.repeatedEnum = [.bar, .baz]
+    }
+    XCTAssertEqual(try msg2.jsonString(options: asStrings), "{\"repeatedEnum\":[\"BAR\",\"BAZ\"]}")
+    XCTAssertEqual(try msg2.jsonString(options: asInts), "{\"repeatedEnum\":[1,2]}")
```

Comment:
I like the way this reads.  Should we bless this by adding some sugar?
```
     let a = try msg.jsonString(options: .enumsAsInts)
```

Reply:
@tbkka if we only ever had booleans, I think we could use `OptionSet`, and maybe it would work out simpler for the initialization code, but since we already have some that aren't, I don't think we can do something nicer.

If we split the bools into an `OptionSet` and everything else as unique fields, then we might have some middle ground, but given that's a code breaking changes, I'm guessing we don't want to consider that?  (And I'm not sure how I like grouping some in an `OptionSet` and others outside it, but all in a `struct` that still needs to be named `Options`.)

Reply:
Makes sense.  Sounds like we should leave it as-is.